assets/src/edit-story/components/panels/textStyle/font.js

```diff
@@ -158,17 +170,67 @@ function FontControls({ selectedElements, pushUpdate }) {
     []
   );
 
+  const fontMap = useMemo(
+    () =>
+      [...fonts, ...recentFonts, ...curatedFonts].reduce(
+        (lookup, option) => ({
+          ...lookup,
+          [option.id]: option,
+        }),
+        {}
+      ),
+    [fonts, recentFonts, curatedFonts]
+  );
+
+  const renderer = ({ option, optionRef, ...rest }) => {
+    return (
+      <Option
+        {...rest}
+        ref={optionRef}
+        fontFamily={
+          option.service.includes('google')
+            ? `'${option.name}::MENU'`
+            : option.name
+        }
+      >
+        {fontFamily === option.id && (
+          <Selected aria-label={__('Selected', 'web-stories')} />
+        )}
+        {option.name}
+      </Option>
+    );
+  };
+
+  const onObserve = (observedFonts) => {
+    ensureMenuFontsLoaded(
+      observedFonts.filter(
+        (fontName) => fontMap[fontName]?.service === 'fonts.google.com'
+      )
+    );
+  };
   return (
     <>
       {fonts && (
         <Row>
-          <FontPicker
+          <AdvancedDropDown
             data-testid="font"
-            aria-label={__('Font family', 'web-stories')}
+            aria-label={__('Edit: Font family', 'web-stories')}
             options={fonts}
-            value={MULTIPLE_VALUE === fontFamily ? '' : fontFamily}
-            placeholder={MULTIPLE_DISPLAY_VALUE}
+            primaryOptions={curatedFonts}
+            primaryLabel={__('Recommended', 'web-stories')}
+            priorityOptions={recentFonts}
+            priorityLabel={__('Recently used', 'web-stories')}
+            selectedId={MULTIPLE_VALUE === fontFamily ? '' : fontFamily}
+            placeholder={
+              MULTIPLE_VALUE === fontFamily
+                ? MULTIPLE_DISPLAY_VALUE
+                : fontFamily
+            }
+            hasSearch
             onChange={handleFontPickerChange}
+            onObserve={onObserve}
+            renderer={renderer}
+            disabled={!fonts?.length}
           />
```

Comment:
Hm, this was actually more compact than expected. But I was thinking that we kept a `<FontPicker />` component behind the scenes, but it was just a shortcut for `<AdvancedDropDown />` with most attributes fixed.

This however seems okay to me, but it could also be moved to its own component - mostly because of the renderer and the overhead in terms of line count required by that.

Reply:
Moved `fontPicker` to a separate component in the `textStyles` panel. LMK if it makes more sense to you now.